packages/flutter/lib/src/painting/_network_image_web.dart

```diff
@@ -118,61 +142,95 @@ class NetworkImage
     // We use a different method when headers are set because the
     // `ui_web.createImageCodecFromUrl` method is not capable of handling headers.
     if (isSkiaWeb || containsNetworkImageHeaders) {
-      final Completer<web.XMLHttpRequest> completer =
-          Completer<web.XMLHttpRequest>();
-      final web.XMLHttpRequest request = httpRequestFactory();
-
-      request.open('GET', key.url, true);
-      request.responseType = 'arraybuffer';
       if (containsNetworkImageHeaders) {
```

Comment:
I like the first suggestion. I don't think the second suggestion is equivalent because there are some cases where if fetching the bytes directly fails, we do not want to try with an <img> tag instead (e.g. if there are headers).